### ISSUE TITLE
winget does not updating app version info

### ISSUE THREAD

**Issue by demios97** [microsoft/winget-cli#2502]
### Brief description of your issue

winget doesn't see actual version of the program after upgrade of the program through winget itself. Reinstalling doesn't help.
<img width="701" alt="Screenshot 2022-09-07 125741" src="https://user-images.githubusercontent.com/53512334/188852498-8c8493de-2666-4643-94ad-928118caaa56.png">
<img width="577" alt="Screenshot 2022-09-07 125903" src="https://user-images.githubusercontent.com/53512334/188852510-2eea220c-2e7d-4834-a966-aa68d6d0c244.png">

### Steps to reproduce

>>winget install AIMP.AIMP
>>winget upgrade --all

### Expected behavior

After upgrade winget should show me actual version number.

### Actual behavior

Every time I try to upgrade, aimp install himself, no matter the version.

### Environment

```shell
Windows Package Manager v1.3.2091
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19044.1949
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.18.2091.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by Trenly** [comment]
What is shown when you run `winget list AIMP` ?

**Comment by Trenly** [comment]
As I was looking into this, in additon to the bug above, I found that this application installs side by side. 

Setting `UpgradeBehavior: uninstallPrevious` is not an option, as the application uninstaller returns success immediately and is interactive, thus requiring #1885 to provide custom silent uninstall arguments.

**Comment by demios97** [comment]
> What is shown when you run `winget list AIMP` ?

<img width="680" alt="Screenshot 2022-09-08 110837" src="https://user-images.githubusercontent.com/53512334/189070147-c1da68c6-534f-4e7e-a6da-a3af6c7d6c96.png">

**Comment by Trenly** [comment]
> > What is shown when you run `winget list AIMP` ?
> 
> <img alt="Screenshot 2022-09-08 110837" width="680" src="https://user-images.githubusercontent.com/53512334/189070147-c1da68c6-534f-4e7e-a6da-a3af6c7d6c96.png">

Thank you; This looks exactly like the issue I reported in the issue linked above, which means you  don’t have the other side-by-side upgrade issue

**Comment by emoxam** [comment]
Face the same with a bunch of apps
dbeaver.dbeaver
Docker.DockerDesktop
Microsoft.WindowsTerminal
PostgreSQL.pgAdmin
Telegram.TelegramDesktop
Python.Python.3.12
MongoDB.Compass.Full
VideoLAN.VLC

**Comment by emoxam** [comment]
winget update #Refreshes the repositories
winget upgrade #Actually updates apps
is it true ? Why does "winget update" works if we don't have any mention at "winget -?" ?


winget upgrade
```
Имя                                                   ИД                           Версия        Доступно      Источник
-----------------------------------------------------------------------------------------------------------------------
OpenWebStart 1.7.0                                    karakun.OpenWebStart         1.7.0         1.9.0         winget
MongoDB Compass                                       MongoDB.Compass.Full         1.40.3        1.42.0.0      winget
Notepad++ (64-bit x64)                                Notepad++.Notepad++          8.2           8.6.2         winget


pgAdmin 4 version 7.3                                 PostgreSQL.pgAdmin           7.3           8.2           winget
```

winget upgrade PostgreSQL.pgAdmin
```
Найдено pgAdmin 4 [PostgreSQL.pgAdmin] Версия 8.2
Лицензия на это приложение предоставлена вам владельцем.
Корпорация Майкрософт не несет ответственность за сторонние пакеты и не предоставляет для них никакие лицензии.
Скачивание https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v8.2/windows/pgadmin4-8.2-x64.exe
  ██████████████████████████████   181 MB /  181 MB
Хэш установщика успешно проверен
Запуск установки пакета...
Успешно установлено
```
And again

winget upgrade
```
Имя                                                   ИД                           Версия        Доступно      Источник
-----------------------------------------------------------------------------------------------------------------------
OpenWebStart 1.7.0                                    karakun.OpenWebStart         1.7.0         1.9.0         winget
MongoDB Compass                                       MongoDB.Compass.Full         1.40.3        1.42.0.0      winget
Notepad++ (64-bit x64)                                Notepad++.Notepad++          8.2           8.6.2         winget
pgAdmin 4 version 7.3                                 PostgreSQL.pgAdmin           7.3           8.2           winget

```

Why does PostgreSQL.pgAdmin still shows as not updated ?

**Comment by denelon** [comment]
WinGet 1.8 has better support for packages that get installed "side-by-side".
Is this a situation where multiple versions are installed?

If it's not, it's possibly a case of the registry entry not getting updated by the installer.

**Comment by demios97** [comment]
There were no multiple versions of the same app. But the issue was gone long ago. Maybe @emoxam can provide recent feedback?

**Comment by denelon** [comment]
I'm going to go ahead and close this issue. If the problem persists, please re-open this issue or file a new one. I'm fairly confident the issue has been resolved in WinGet 1.8.
